### PR TITLE
Add `cilium-servicemonitors` app

### PR DESCRIPTION
**Pull request by mcharriere**
Signed-off-by: Matias Charriere <matias@giantswarm.io>

### What this PR does / why we need it

Towards https://github.com/giantswarm/roadmap/issues/3283

### Checklist

- [x] Update changelog in CHANGELOG.md.

### Trigger e2e tests

/run cluster-test-suites

### CODE REVIEW

File: helm/default-apps-aws/values.yaml [hunk 1]
```diff
@@ -132,6 +132,19 @@ apps:
...
+    chartName: cilium-servicemonitors
+    catalog: default
+    enabled: true
+    clusterValues:
+      configMap: false
+      secret: false
+    dependsOn: prometheus-operator-crd
+    namespace: kube-system
+    # used by renovate
+    # repo: giantswarm/chart-operator-extensions
```
Comment (mcharriere) [inline helm/default-apps-aws/values.yaml:146]:
```suggestion
    # repo: giantswarm/cilium-servicemonitors-app
```

**Comment by mcharriere** [comment]
I had to help the tests a bit to pass, cluster deletion is still broken.

**Review by kopiczko — APPROVED** [review]

**Pull request merged**